ahorcado.rb

```diff
@@ -1,62 +1,44 @@
-#TODO: word_spaces should depend from current_word
-# Array equivalent to the characters of the word 
-# [“”, “”, “”, “”, ...]
-# []
-# Array.new
-
-#current_word = "love"
-#word_spaces = Array.new(current_word.length, "")
-
-@current_word = "friends"
-@word_spaces = Array.new(@current_word.length, "")
-@current_characters = @current_word.split("")
-
-def display_game
-  @word_spaces.each do |characters|
-    print "| #{characters} | "
+class Ahorcado
+  def initialize(current_word)
+    @current_word = current_word
+    @word_spaces = Array.new(current_word.length, "")
+    @current_characters = current_word.split("")
+
+    display_game
+  end
+
+  def display_game
+    @word_spaces.each do |characters|
+      print "| #{characters} | "
+    end
+    choose_character
   end
-  choose_character
-end
 
-def word_spaces_empty?
-  @word_spaces.all? { |w| w == "" }
-end
+  def word_spaces_empty?
+    @word_spaces.any? { |w| w == "" }
+  end
 
-def choose_character
-  while word_spaces_empty? do
-    puts "Guess the word! Choose a letter"
-    chosen_character = gets.chomp
+  def choose_character
+    while word_spaces_empty? do
```

Comment:
The naming here sounds weird. Could you think in a better name that sounds better?

` while [SOMETHING IS HAPPENING] do`

Don't worry about being a bit verbose. Most of the times, that's welcomed.

Reply:
Sure, I will change it 💃